x/programs/rust/sdk_macros/src/lib.rs

```diff
@@ -46,19 +46,41 @@ pub fn public(_: TokenStream, item: TokenStream) -> TokenStream {
         }
         panic!("Unsupported function parameter format.");
     });
-
     let (param_names, param_types): (Vec<_>, Vec<_>) = full_params.unzip();
+
+    let program_param = &param_names[0];
+    // convert to a string
+    let program_name = name.to_string();
```

Comment:
nit: this is the function name isn't it?

Reply:
yep

Reply:
```suggestion
    let function_name = name.to_string();
```
?